### PR TITLE
rename, restructure, and reorganize timeout-related types

### PR DESCRIPTION
**Pull request by Velfi**
## Motivation and Context

This PR removes currently unused fields in `TimeoutConfig` and refactors the way timeouts are passed to different parts of the network stack. It also changes the names of several structs and functions related to timeouts.

## Description

rename: aws_smithy_types::timeoutTimeoutConfig to SharedTimeoutConfig
rename: aws_smithy_client::timeout::Settings to HttpConnectorTimeoutConfig
rename: aws_smithy_client::timeout::ClientTimeoutParams to ServiceClientTimeoutConfig
remove: currently unimplemented timeouts from SharedTimeoutConfig
update: don't set pointless timeout config during TokenMiddleware creation
update: don't set pointless timeout config during IMDS client creation
rename: ProfileFileTimeoutConfigProvider to ProfileFileSharedTimeoutConfigProvider
remove: currently unimplemented timeouts from ProfileFileSharedTimeoutConfigProvider
rename: EnvironmentVariableTimeoutConfigProvider to EnvironmentVariableSharedTimeoutConfigProvider
remove: currently unimplemented timeouts from EnvironmentVariableSharedTimeoutConfigProvider
rename: aws_config::profile::timeout_config to aws_config::profile::shared_timeout_config
rename: aws_config::environment::timeout_config to aws_config::environment::shared_timeout_config
rename: fn generate_timeout_service_params_from_timeout_config to service_client_timeout_config_from_shared_timeout_config
update: HyperAdapter Builder now takes HttpConnectorTimeoutConfig
update: HttpCredentialProvider Builder now takes HttpConnectorTimeoutConfig
update: HttpSettings now wraps HttpConnectorTimeoutConfig instead of SharedTimeoutConfig

## Testing

ran existing tests

## Checklist

- [ ] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [ ] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Velfi** [comment]
I haven't updated the changelog yet but I wanted to put this up for review anyways

File: aws/rust-runtime/aws-config/src/environment/shared_timeout_config.rs [hunk 1]
```diff
@@ -5,61 +5,60 @@
...
+};
 use aws_types::os_shim_internal::Env;
 use std::time::Duration;
 
-const ENV_VAR_CONNECT_TIMEOUT: &str = "AWS_CONNECT_TIMEOUT";
-const ENV_VAR_TLS_NEGOTIATION_TIMEOUT: &str = "AWS_TLS_NEGOTIATION_TIMEOUT";
-const ENV_VAR_READ_TIMEOUT: &str = "AWS_READ_TIMEOUT";
+// const ENV_VAR_CONNECT_TIMEOUT: &str = "AWS_CONNECT_TIMEOUT";
+// const ENV_VAR_TLS_NEGOTIATION_TIMEOUT: &str = "AWS_TLS_NEGOTIATION_TIMEOUT";
+// const ENV_VAR_READ_TIMEOUT: &str = "AWS_READ_TIMEOUT";
```
Comment (rcoh) [inline aws/rust-runtime/aws-config/src/environment/shared_timeout_config.rs:16]: should probably note why these are commented out, maybe with link to a github issue

File: rust-runtime/aws-smithy-types/src/timeout.rs [hunk 2]
```diff
@@ -34,39 +34,22 @@ use std::time::Duration;
...
-    connect_timeout: Option<Duration>,
-    tls_negotiation_timeout: Option<Duration>,
-    read_timeout: Option<Duration>,
+pub struct SharedTimeoutConfig {
+    // connect_timeout: Option<Duration>,
+    // tls_negotiation_timeout: Option<Duration>,
+    // read_timeout: Option<Duration>,
     api_call_attempt_timeout: Option<Duration>,
     api_call_timeout: Option<Duration>,
 }
```
Comment (rcoh) [inline rust-runtime/aws-smithy-types/src/timeout.rs:43]: so this is timeouts on operations? maybe it should be something like `ApiTimeoutConfig`?

Comment (rcoh) [reply rust-runtime/aws-smithy-types/src/timeout.rs:43]: will there eventually be a `RootTimeoutConfig` that has both `ApiTimeoutConfig` and `HttpSettingsTimeoutConfig`?

**Review by rcoh — COMMENTED** [review]
not totally sure about `SharedTimeoutConfig` name-wise, but overall this seems like the right direction

**Comment by Velfi** [comment]
closing in favor of #1246

**Pull request closed without merging**